syntax = "proto3";

option go_package = "api-go/src/service/gateway";  // プロジェクトのルートからの相対パス

service ChartService {
  rpc GenerateChart(ChartRequest) returns (ChartResponse) {}
}

message ChartRequest {
  repeated float stock_data = 1;  // 株価データ
  repeated float indicator_data = 2;  // 指標データ
  string ticker = 3;  // ティッカーシンボル
}

message ChartResponse {
  string chart_data = 1;  // Base64エンコードされた画像データ
}


// python -m grpc_tools.protoc -I src/generate_chart --python_out src/generate_chart --grpc_python_out=src/generate_chart src/generate_chart/generate_chart.proto